common/djangoapps/student/roles.py

```diff
@@ -153,7 +153,8 @@ def add_users(self, *users):
         # legit get updated.
         from student.models import CourseAccessRole
         for user in users:
-            if user.is_authenticated and user.is_active and not self.has_user(user):
+            user = User.objects.get(id=user.id)
```

Comment:
Why is this additional query necessary? This seems like a step backwards.

Sorry for the many back-and-forth comments here :-/